jumanji/environments/routing/pacman/env.py

```diff
@@ -0,0 +1,551 @@
+        ps = jnp.array([player_space.y, player_space.x])
+
+        # Get the number of pellets on the map
+        num_pellets = state.pellets
+
+        # Check if player has eaten a pellet in this step
+        ate_pellet = jnp.any(jnp.all(ps == pellet_spaces, axis=-1))
+
+        # Reduce number of pellets on map if eaten, add reward and remove eaten pellet
+        num_pellets -= ate_pellet.astype(int)
+        rewards = ate_pellet * 10.0
+        mask = jnp.logical_not(jnp.all(ps == pellet_spaces, axis=-1))
+        pellet_spaces = pellet_spaces * mask[..., None]
+
+        return rewards, pellet_spaces, num_pellets
+
+    def player_step(self, state: State, action: int, steps: int = 1) -> Position:
+        """
+        Compute the new position of the player based on the given state and action.
+
+        Args:
+            state: 'state` object corresponding to the new state of the environment.
+            action: an integer between 0 and 4 representing the player's chosen action.
+            steps: how many steps ahead of current position to search.
+
+        Returns:
+            new_pos: a `Position` object representing the new position of the player.
+        """
+
+        position = state.player_locations
+
+        move_left = lambda position: (position.y, position.x - steps)
+        move_up = lambda position: (position.y - steps, position.x)
+        move_right = lambda position: (position.y, position.x + steps)
+        move_down = lambda position: (position.y + steps, position.x)
+        no_op = lambda position: (position.y, position.x)
+
+        new_pos_row, new_pos_col = jax.lax.switch(
+            action, [move_left, move_up, move_right, move_down, no_op], position
+        )
+
+        new_pos = Position(x=new_pos_col % self.x_size, y=new_pos_row % self.y_size)
+        return new_pos
+
+    def check_power_up(
+        self, state: State
+    ) -> Tuple[chex.Array, chex.Numeric, chex.Numeric]:
+        """
+        Check if the player is on a power-up location and update the power-up
+        locations array accordingly.
+
+        Args:
+            state: 'state` object corresponding to the new state of the environment
+
+        Returns:
+            power_up_locations: locations of the remaining power-ups
+            eat: a bool indicating if the player can eat the ghosts
+            reward: an int of the reward gained from collecting power-ups
+        """
+
+        power_up_locations = jnp.array(state.power_up_locations)
+
+        player_space = state.player_locations
+        player_loc_x = player_space.x
+        player_loc_y = player_space.y
+        player_loc = jnp.array([player_loc_y, player_loc_x])
+
+        # Check if player and power_up position are shared
+        on_powerup = (player_loc == power_up_locations).all(axis=-1).any()
+        eat = on_powerup.astype(int)
+        mask = (player_loc == power_up_locations).all(axis=-1)
+        invert_mask = mask != ~mask
```

Comment:
Should this not just be?

```suggestion
        invert_mask = ~mask
```